ext/opentelemetry-ext-jaeger/src/opentelemetry/ext/jaeger/__init__.py

```diff
@@ -0,0 +1,394 @@
+
+    :param collector_host_name: (Optional) The host name of the Jaeger-
+                                Collector HTTP Thrift.
+
+    :param collector_port: (Optional) The port of the Jaeger-Collector HTTP
+                           Thrift.
+
+    :param username: (Optional) The user name of the Basic Auth
+                     if authentication is required.
+
+    :param password: (Optional) The password of the Basic Auth
+                     if authentication is required.
+
+    :param collector_endpoint: (Optional) The endpoint of the Jaeger-Collector
+                               HTTP Thrift.
+
+    :param agent_host_name: (Optional) The host name of the Jaeger-Agent.
+
+    :param agent_port: (Optional) The port of the Jaeger-Agent.
+    """
+
+    def __init__(
+        self,
+        service_name: str = "my_service",
+        agent_host_name: str = DEFAULT_AGENT_HOST_NAME,
+        agent_port: int = DEFAULT_AGENT_PORT,
+        collector_host_name: str = None,
+        collector_port: int = None,
+        collector_endpoint: str = DEFAULT_COLLECTOR_ENDPOINT,
+        username: str = None,
+        password: str = None,
+    ):
+        self.service_name = service_name
+        self.agent_host_name = agent_host_name
+        self.agent_port = agent_port
+        self._agent_client = None
+        self.collector_host_name = collector_host_name
+        self.collector_port = collector_port
+        self.collector_endpoint = collector_endpoint
+        self.username = username
+        self.password = password
+        self._collector = None
+
+    @property
+    def agent_client(self):
+        if self._agent_client is None:
+            self._agent_client = AgentClientUDP(
+                host_name=self.agent_host_name, port=self.agent_port
+            )
+        return self._agent_client
+
+    @property
+    def collector(self):
+        if self._collector is not None:
+            return self._collector
+
+        if self.collector_host_name is None or self.collector_port is None:
+            return None
+
+        thrift_url = "http://{}:{}{}".format(
+            self.collector_host_name,
+            self.collector_port,
+            self.collector_endpoint or DEFAULT_COLLECTOR_ENDPOINT,
+        )
+
+        auth = None
+        if self.username is not None and self.password is not None:
+            auth = (self.username, self.password)
+
+        self._collector = Collector(thrift_url=thrift_url, auth=auth)
+        return self._collector
+
+    def export(self, spans: typing.Sequence[Span]):
+        """
+        :param spans:
+            Tuple of spans to export.
+        """
+        jaeger_spans = self.translate_to_jaeger(spans)
+
+        batch = jaeger.Batch(
+            spans=jaeger_spans,
+            process=jaeger.Process(serviceName=self.service_name),
+        )
+
+        if self.collector is not None:
+            self.collector.submit(batch)
+        self.agent_client.emit(batch)
```

Comment:
According to my understanding from OpenCensus" the agent is mandatory and the collector is optional, i.e, spans are only sent to the collector when it's configured. I asked for further clarification on gitter because it is also not that clear for me.

Reply:
FYI the original author @sonmezonur put this code in https://github.com/census-instrumentation/opencensus-python/pull/154/files.

Reply:
Just to add some more info to this discussion.
- in otel-go the expoter sends data to only one, Collector of Agent, the collector has higher priority: https://github.com/open-telemetry/opentelemetry-go/blob/7797bf983775e7a95f7504468598f3cd1044e8a4/exporter/trace/jaeger/jaeger.go#L331
- AFAIU in otel-java only the collector is used: https://github.com/open-telemetry/opentelemetry-java/blob/master/exporters/jaeger/src/main/java/io/opentelemetry/exporters/jaeger/JaegerGrpcSpanExporter.java#L123

I'd rather prefer to keep the OpenCensus behavior (always send to Agent and optionally send to Collector if configured) if there are not objections.